datafusion/common/src/dfschema.rs

```diff
@@ -203,7 +203,20 @@ impl DFSchema {
                 // qualifier and name.
                 (Some(q), Some(field_q)) => q == field_q && field.name() == name,
                 // field to lookup is qualified but current field is unqualified.
-                (Some(_), None) => false,
+                (Some(qq), None) => {
+                    // the original field may now be aliased with a name that matches the
+                    // original qualified name
+                    let table_ref: TableReference = field.name().as_str().into();
+                    match table_ref {
+                        TableReference::Partial { schema, table } => {
+                            schema == qq && table == name
+                        }
+                        TableReference::Full { schema, table, .. } => {
+                            schema == qq && table == name
+                        }
+                        _ => false,
+                    }
```

Comment:
I'm not crazy about this but I don't see how else we can handle this unless we introduce some sort of alias-with-qualifier concept

Reply:
Perhaps aliases should be `Column` instead of `String`?

```rust
enum Expr {
  Alias(expr: Box<Expr>, alias: Column>
  ...
  }
```

Reply:
I don't think alias's can be fully qualified identifiers

I think the core disconnect is that datafusion 'flattens' compound identifiers into strings for column names which can then be referred to by anything that reads the plan that makes them

Postgres deals with this type of issue by using indexes to identify columns, but that ends up with its own problems as the indexes invariably get mixed up sometimes and debugging it is super tricky.